docs/blueprints/zombie-segments-cleaner.md

```diff
@@ -0,0 +1,55 @@
+# Zombie Segments Cleaner
+
+## Abstract
+
+This document describes a design for cleaning segments that are not accessible with standard operations like listing.
+
+## Background
+
+Currently with a multisegment object, we have several places where we can leave unfinished or broken objects:
+* if upload process will be stopped, because of an error or canceling, when one or more segments are already uploaded we can become zombie segments.
+* if delete process will be stopped, because of an error or canceling, when one or more segments are already deleted we can have an object with an incomplete number of segments.
+* in the past we had a bug where error during deleting segments from nodes was interrupting deleting segment from satellite, and we have segments available on satellite but not on storage node.
+* deleted bucket can leave all objects inaccessible ???
+* deleted project can leave all objects inaccessible ???
+
+We need a system for identifying objects with missing segments, and zombie segments that are not listed as part of objects, that cleans up those segments.
+
+In the long term, the metainfo refactor will fix this. We need a short term solution.
+
+## Design
+
+How detect zombie segment:
+* segment doesn't have corresponding last segment (`l`)
```

Comment:
We have cases where we are missing `l`, but every other segment exists https://github.com/storj/storj/blob/master/satellite/metainfo/metainfo.go#L1150

Reply:
you mean that we have this window where before making the last segment we have several valid segments?

Reply:
I mean we have valid segments that don't have a corresponding last segment.

Reply:
I'm not sure I'm following use case for `CompareAndSwap` here. In `CommitObject` we are moving (delete and put) pointer from `sN` segment to `l` segment.

Reply:
As I understand, until the final `Put` there's a period where `l` segment doesn't exist? Or am I mistaken?

Reply:
Yes, exactly

Reply:
I will add this special case to doc but I'm thinking that if this is only a temporary solution we can delete only segments older than e.g. 1 week.